packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -102,16 +113,68 @@ export interface MetricExporter {
 /**
  * Base interface for aggregators. Aggregators are responsible for holding
  * aggregated values and taking a snapshot of these values upon export.
+ *
+ * Use {@link Aggregator} instead of this BaseAggregator.
  */
-export interface Aggregator {
+interface BaseAggregator {
```

Comment:
No reason to change aggregator
```
export interface Aggregator {
  kind: AggregatorKind;
  /** Updates the current with the new value. */
  update(value: number): void;

  /** Returns snapshot of the current point (value with timestamp). */
  toPoint(): Point<PointValueType>;
}

export interface Point<T> {
  value: T;
  timestamp: HrTime;
}

```

And then you don't need to create a separate interfaces for all different kind of metrics. Just use 
```
export class MyAggregator implements Aggregator {
  private _distribution: Distribution;
  kind = AggregatorKind.DISTRIBUTION;
  toPoint(): Point<Distribution> {
    return {
      value: this._distribution,
      timestamp: [0, 0]
    }
  }
  update(value: number) {
....
  }
}

const a = new MyAggregator();
// and now easy
a.toPoint().value.last;

```

Although I think that this change with metrics should be done first as one smaller PR

Reply:
The change made to Aggregator is that one aggregator is determined to return one type of point value type.

The previous Aggregator definition and your suggested one made it possible to return any one of the point value type in single aggregator, i.e. an aggregator can return a LastValue and a Histogram for each `toPoint` call since they all conform to the signature.

The solution proposed in the PR made an aggregator can only return a definite subset of PointValueType. i.e. an aggregator with HistogramAggregatorType returning a LastValue is not going to pass the type check. This made the aggregator kind check prominent and enforced.

Reply:
As you suggested, I'm going to move this part of the changes to a separate PR.